### PR TITLE
sync deleted capabilities when `vela system update`

### LINKED ISSUE #128: [BUG] vela refresh doesn't synchronized deleted workloaddefinition
**Describe the bug**

vela refresh doesn't synchronized deleted workloaddefinition

**To Reproduce**

- delete workloaddefinition `containerizedworkloads.core.oam.dev`
- run `vela -h`
```
  Workloads:
    containerized:run <appname> [args]	Run containerized workloads
    deployment:run <appname> [args]   	Run deployment workloads
```
- run `vela refresh`
```
[root@iZj6cc54ecpc3pm8t9yctuZ Rudrx]# vela refresh
syncing workload definitions from cluster...
get 0 workload definitions from cluster, syncing to /root/.vela/definitions/workloads...0 workload definitions successfully synced
syncing trait definitions from cluster...
get 2 trait definitions from cluster, syncing to /root/.vela/definitions/traits...2 trait definitions successfully synced
```

- run `vela -h`
```
  Workloads:
    containerized:run <appname> [args]	Run containerized workloads
    deployment:run <appname> [args]   	Run deployment workloads
```

**Expected behavior**

command ` containerized:run` should not display after deletion.

### PR DESCRIPTION
**Pull request by captainroy-hy**
fix #128 

Signed-off-by: roy wang <seiwy2010@gmail.com>

### CODE REVIEW

**Review by wonderflow — COMMENTED** [review]
Good job, please add test for this

**Comment by captainroy-hy** [comment]
yes,got it.

File: pkg/commands/refresh.go [hunk 1]
```diff
@@ -52,8 +54,12 @@ func RefreshDefinitions(ctx context.Context, c client.Client, ioStreams cmdutil.
 	if err != nil {
 		return err
 	}
+	syncedTemplates = append(syncedTemplates, templates...)
 	ioStreams.Infof("get %d trait definitions from cluster, syncing...", len(templates))
 	successNum = plugins.SinkTemp2Local(templates, dir)
 	ioStreams.Infof("%d trait definitions successfully synced\n", successNum)
+
+	legacyNum := plugins.RemoveLegacyTemps(syncedTemplates, dir)
+	ioStreams.Infof("%d legacy capability definitions successfully removed\n", legacyNum)
```
Comment (zzxwill) [inline pkg/commands/refresh.go:63]: If `legacyNum` is equals to zero, the successful message will seem odd.

Comment (captainroy-hy) [reply pkg/commands/refresh.go:63]: yes, I will fix it.

File: pkg/plugins/local_test.go [hunk 2]
```diff
@@ -103,3 +106,42 @@ func testInDir(t *testing.T, casename, dir string, tmps, defexp []types.Capabili
...
+		func(t *testing.T) {
+			err := os.MkdirAll(dir, 0755)
+			assert.NoError(t, err, c.caseName)
+			defer os.RemoveAll(dir)
+			existingTemps := []types.Capability{deployment, statefulset, route}
+			number := SinkTemp2Local(existingTemps, dir)
+			assert.Equal(t, 3, number)
+			rmNum := RemoveLegacyTemps(c.newTemps, dir)
+			assert.Equal(t, c.rmNum, rmNum, c.caseName)
+		}(t)
```
Comment (wonderflow) [inline pkg/plugins/local_test.go:145]: writing a function inside the test loop makes  it hard to read, I suggest to separate it out, or don't use defer.

Comment (captainroy-hy) [reply pkg/plugins/local_test.go:145]: got it.

**Review by wonderflow — APPROVED** [review]

**Pull request merged**